src/Microsoft.OData.Client/IHttpClientProvider.cs

```diff
@@ -9,21 +9,19 @@
 namespace Microsoft.OData.Client
 {
     /// <summary>
-    /// Provides the <see cref="HttpClientHandler"/> instance(s) used by OData Client
+    /// Provides the <see cref="HttpClient"/> instance(s) used by OData Client
     /// to execute requests.
     /// </summary>
-    public interface IHttpClientHandlerProvider
+    public interface IHttpClientProvider
     {
         /// <summary>
-        /// Returns an <see cref="HttpClientHandler"/> instance that OData Client should
+        /// Returns an <see cref="HttpClient"/> instance that OData Client should
         /// used to execute a request. This method is called per request.
         /// </summary>
-        /// <returns>The <see cref="HttpClientHandler"/> instance that will be used to execute the request.</returns>
+        /// <returns>The <see cref="HttpClient"/> instance that will be used to execute the request.</returns>

```

Comment:
Why not use IHttpClientFactory ?

Reply:
Good question. One concern with `IHttpClientFactory` is .NET Framework support which we have not yet finalized a decision on. Another concern is whether requesting `IHttpClientFactory` would be a barrier to people who are using OData client in an environment where they don't have an instance of `IHttpClientFactory`.

Reply:
I may have been wrong about the .NET Framework support. `IHttpClientFactory` is defined in .NET Platform Extensions and not in the core standard lib. And according to this chart on NuGet, it seems to be supported in .NET FX: https://www.nuget.org/packages/Microsoft.Extensions.Http#supportedframeworks-body-tab